spyder/utils/programs.py

```diff
@@ -710,7 +712,27 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                 else:
                     run_program(program['cmd'], arglist)
                 return
-        # TODO: Add a fallback to OSX
+    elif sys.platform == 'darwin':
+        f = tempfile.NamedTemporaryFile('wt', suffix='.sh',
+                                        dir=get_temp_dir(), delete=False)
+        f.write(' '.join(p_args))
+        f.close()
+        os.chmod(f.name, 0o777)
+
+        def run_terminal_thread():
+            if wdir:
+                proc = subprocess.Popen(

```

Comment:
I think we should use here the function `run_shell_command` (present in this same module), instead of `subprocess` directly. Please try that to see if it works.